rfcs/0013-deprecate-currency-factory.md

```diff
@@ -0,0 +1,93 @@
+# Depreciation of Currency Factory - Assessment RFC
+
+This assessment aims to gather all relevant details and consequences of 
+depreciating Currency Factory and its replacements.
+
+## Reasoning
+
+There are several reasons we stand to benefit from replacing Currency Factory.
+These reasons are detailed below:
+
+### Currency Factory Ranges are Confusing
+  
+As detailed in [this](https://composablefinance.slack.com/archives/C031G5NT0CA/p1667492928188269) 
+thread, the Currency Factory ranges are a confusing abstraction that we don't 
+stand to gain from. We really only have two types of currencies native 
+(mintable) and non-native (external). However, we have many more ranges that 
+enforce artificial limitations on various abstractions of these two currency 
+types. This complicates the currency creation process and leads to confusion 
+while providing no benefit to us.
+
+### Currency Factory is Already Losing Functionality
+
+Recently, we moved away from storing ED in Currency Factory. This leaves two 
+uses for currency factory: reserving asset IDs and storing asset metadata. 
+
+As we have already discussed, currency factory over complicates the asset ID 
+reservation system. As for asset metadata, we don't have standards for 
+formatting this nor do we utilize it.
+
+Assuming we replace the reservation system, Currency Factory will be left 
+without value.
+
+## Solution Proposals
+
+There are multiple ways we could go about replacing Currency Factory. These 
+solutions are detailed below:
+
+### Simplify & Move the Asset ID Reservation System
+
+One way to remove the need for pallet-currency-factory is to move its 
+functionality to pallet-asset-registry. If we also simplify the asset ID 
```

Comment:
1. we are 4 month old orml-tokens. what state they have now? may be they did some stuff for non payable assets and substrate assets compatibility?
2. could we move functionally to our assets pallet? can we make our assets pallet to be as compatible as possible with substrate assets?

so in theory we can ditch currency factory and move all to our assets in a way compable on binary level with substrate assets.

also we can throw away our governance-registry imho

Reply:
> we are 4 month old orml-tokens. what state they have now? may be they did some stuff for non payable assets and substrate assets compatibility?

We'll either need to request another back-port or upgrade our runtime to enjoy their newer additions

> could we move functionally to our assets pallet? can we make our assets pallet to be as compatible as possible with substrate assets?

Like you concluded later, I think maintaining a fork would be the easiest way to get all the benefits of Parity's pallet-assets

Reply:
Is our old substrate version holding us back from using newer features? And if so, what is keeping us on an older substrate version?